CHANGELOG.md

```diff
@@ -4,6 +4,127 @@
 
 * Decided for Golang semantic versioning [#208](https://github.com/corazawaf/coraza/issues/208)
 
+* BREAKING: Required Go version gets bumped to 1.18 [#343](https://github.com/corazawaf/coraza/pull/343)
+
+* **CRS v4 Support** - [#218](https://github.com/corazawaf/coraza/issues/218) -
+Coraza v3 is fully compatible with the OWASP Core Rule Set v4. Support for Core Rule Set 3.x is dropped as rules are still based on PCRE. CRS v4 is solely based on RE2 and will soon see a public & official release.  
+
+  CRS further moved away from its dependency on old Regexp::Assemble to the new crs-toolchain helper to prevent rules incompatible with Coraza.
+
+* **FEATURE: TinyGo & WASM support & Envoy Proxy Connector** - [#254](https://github.com/corazawaf/coraza/pull/254) -
+Coraza adds initial support for TinyGo to allow compiling to Web Assembly (WASM). Compatible directives are marked with `Tinygo Compatibility` in the documentation.
+
+  Based on contributions sponsored by the Tetrate.io team, a new [coraza-proxy-wasm connector](https://github.com/corazawaf/coraza-proxy-wasm) is developed which can be loaded in [Envoy Proxy](https://www.envoyproxy.io) or as Istio plugin.
+
+  Special thanks to the Tetrate team, Anuraag Agrawal, Matteo Pace and José Carlos Chávez. This greatly enhances our coverage, as now we support Envoy proxy and any WASM-proxy compatible system.
+
+* **FEATURE: SecDatasets & operators** - [#361](https://github.com/corazawaf/coraza/pull/361) -
+SecDatasets are added as replacement for .data files. WASM support is an essential feature of Coraza v3, but users cannot fully enjoy its potential because of file reading limitations. For this reason, SecDataset is a decent replacement for .data files. It's also easier to watch files for reloading on .conf files.
+
+  Two new SecLang operators are added which can be used to query datasets. `pmFromDataset` and `IpMatchFromDataset` [#75e8217](https://github.com/corazawaf/coraza/commit/75e821700de9fbfafde6c763f474c7add8dab319) can be used instead of their file based equivalents.
```

Comment:
```suggestion
  Two new SecLang operators are added which can be used to query datasets. `pmFromDataset` and `IpMatchFromDataset` [#75e8217](https://github.com/corazawaf/coraza/commit/75e821700de9fbfafde6c763f474c7add8dab319) can be used instead of their file based equivalents for those environments which can't access the filesystem.
```